src/neptune_optuna/impl/__init__.py

```diff
@@ -165,6 +167,7 @@ def __init__(
             (bool, type(None)),
         )
         verify_type("target_names", target_names, (list, type(None)))
+        verify_type("log_all_trials", log_all_trials, (bool, type(None)))
```

Comment:
We cannot provide `None` if I understand :wink:
```suggestion
        verify_type("log_all_trials", log_all_trials, bool)
```